src/app/shared/table-of-contents/table-of-contents.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  Component, Input, AfterViewInit, Inject, ElementRef, OnInit
+} from '@angular/core';
+import {DOCUMENT} from '@angular/platform-browser';
+import {Router, ActivatedRoute, Event, NavigationEnd} from '@angular/router';
+import {Observable} from 'rxjs/Observable';
+import {Subscription} from 'rxjs/Subscription';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/operator/debounceTime';
+
+interface Link {
```

Comment:
Add some descriptions on this type? Not immediately obvious exactly what `type`, `top` etc. refer to or how `Link` in general is meant to be used